### PR TITLE
fix(evm): prevent nil pointer access

### PR DESCRIPTION
**Pull request by dessaya**

### CODE REVIEW

File: packages/vm/core/evm/evmtest/setup.go [hunk 1]
```diff
@@ -263,7 +259,7 @@ func (e *SoloChainEnv) maxGasLimit() uint64 {
 }
 
 func (e *SoloChainEnv) DeployContract(creator *ecdsa.PrivateKey, abiJSON string, bytecode []byte, args ...interface{}) *EVMContractInstance {
```
Comment (jorgemmsilva) [inline packages/vm/core/evm/evmtest/setup.go:261]: makes sense to add the value as a parameter (might be used by some contracts)

File: packages/vm/core/evm/evmimpl/impl.go [hunk 2]
```diff
@@ -424,7 +430,9 @@ func callContract(ctx isc.Sandbox) dict.Dict {
 		ctx.Privileged().GasBurnEnable(true)
 		gasErr := panicutil.CatchPanic(
 			func() {
-				ctx.Gas().Burn(gas.BurnCodeEVM1P, gas.EVMGasToISC(res.UsedGas, &gasRatio))
+				if res != nil {
+					ctx.Gas().Burn(gas.BurnCodeEVM1P, gas.EVMGasToISC(res.UsedGas, &gasRatio))
```
Comment (jorgemmsilva) [inline packages/vm/core/evm/evmimpl/impl.go:434]: when can `res` be `nil`? do we not charge gas for this tx?

File: packages/vm/core/evm/evmtest/setup.go (hunk 1 above)
Comment (dessaya) [reply packages/vm/core/evm/evmtest/setup.go:261]: Yes, I needed this for Chris' contract that had a `constructor() payable`.

File: packages/vm/core/evm/evmimpl/impl.go (hunk 2 above)
Comment (dessaya) [reply packages/vm/core/evm/evmimpl/impl.go:434]: It happened to me when sending a tx with a very small gasLimit, less than the intrinsic gas.

I guess we already charge the minimum gas set in ISC? I'll add a test.

Comment (jorgemmsilva) [reply packages/vm/core/evm/evmimpl/impl.go:434]: we already handle this in the jsonrpc side:
https://github.com/iotaledger/wasp/blob/develop/packages/evm/jsonrpc/jsonrpctest/env.go#L363-L373
https://github.com/iotaledger/wasp/blob/develop/packages/evm/jsonrpc/waspevmbackend.go#L59-L65

not sure if we should add this `res != nil` check here. could potentially be exploited by some edge case that makes res be empty? :shrug:

Comment (dessaya) [reply packages/vm/core/evm/evmimpl/impl.go:434]: I just added the test, without the `res != nil` I get nil pointer. Probably because I'm calling from solo instead of jsonrpc.

I don't see how this could add a vulnerability that wasn't there in the first place? The behavior is still the same, the only thing that changes is the error reported ("intrinsic gas" instead of "nil pointer").

Comment (jorgemmsilva) [reply packages/vm/core/evm/evmimpl/impl.go:434]: You're right. Disregard my last comment :+1:

**Review by jorgemmsilva — APPROVED** [review]
